File: portfolioblog.client/src/models/blog-article/BlogArticle.ts
import { ArticleCategories } from "./ArticleCategories";
import { CalendarMonths } from "./CalendarMonths";

export interface BlogArticle {
    id: number;
    encryptedId: string | null; // Corresponds to nullable EncryptedId
    title: string;
    articleBody: string;
    photoPath: string;
    year: number;
    month: CalendarMonths;
    day: number;
    rank: number;
    category: ArticleCategories;
}

File: PortfolioBlog.Server/Models/Blog/Articles/BlogArticle.cs
using System.ComponentModel.DataAnnotations.Schema;
using PortfolioBlog.Server.Models.Blog.Articles.ArticleSubEnums;

namespace PortfolioBlog.Server.Models.Blog.Articles
{
    public class BlogArticle
    {
        public int Id { get; set; }
        [NotMapped]
        public string EncryptedId { get; set; }
        public string Title { get; set; }
        public string ArticleBody { get; set; }
        public string PhotoPath { get; set; }

        public int Year { get; set; }
        public CalendarMonths Month { get; set; }
        public int Day { get; set; }
        public double Rank { get; set; }
        public ArticleCategories Category { get; set; }
    }
}
